src/sip/transp.c

```diff
@@ -299,6 +311,13 @@ static void sip_recv(struct sip *sip, const struct sip_msg *msg,
 		if (msg->req != lsnr->req)
 			continue;
 
+		if (msg->req) {
+			if (!have_essential_fields(msg)){
```

Comment:
Is it maybe enough to check this once before the while loop?

Reply:
Sounds good, checking once is enough since the msg struct stays const inside the while loop. Thanks for pointing that out.